From 101acd9bde13bdcfef95a89a94d7836bbc39ae72 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Wed, 19 Jun 2019 16:19:44 -0400
Subject: [PATCH 14/35] acpi: Ignore acpi_rsdp kernel param when the kernel has
 been locked down

Message-id: <20190619162005.32322-15-labbott@redhat.com>
Patchwork-id: 265667
O-Subject: [ARK PATCH 14/35] acpi: Ignore acpi_rsdp kernel param when the kernel has been locked down
Bugzilla:
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

From: Josh Boyer <jwboyer@redhat.com>

This option allows userspace to pass the RSDP address to the kernel, which
makes it possible for a user to modify the workings of hardware .  Reject
the option when the kernel is locked down.

Signed-off-by: Josh Boyer <jwboyer@redhat.com>
Signed-off-by: David Howells <dhowells@redhat.com>
cc: Dave Young <dyoung@redhat.com>
cc: linux-acpi@vger.kernel.org
Signed-off-by: Matthew Garrett <matthewgarrett@google.com>
---
 drivers/acpi/osl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 drivers/acpi/osl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/acpi/osl.c b/drivers/acpi/osl.c
index cc7507091dec..e4801aeeaadc 100644
--- a/drivers/acpi/osl.c
+++ b/drivers/acpi/osl.c
@@ -180,7 +180,7 @@ acpi_physical_address __init acpi_os_get_root_pointer(void)
 	acpi_physical_address pa;
 
 #ifdef CONFIG_KEXEC
-	if (acpi_rsdp)
+	if (acpi_rsdp && !kernel_is_locked_down("ACPI RSDP specification"))
 		return acpi_rsdp;
 #endif
 	pa = acpi_arch_get_root_pointer();
-- 
2.21.0

